tests/unit/plugins/test_rust_plugin.py

```diff
@@ -262,6 +262,78 @@ def callback_fail():
     ]
 
 
+@pytest.mark.parametrize("after", [["something-else"], []])
+def test_validate_environment_should_have_rustup(
+    after, fake_process: pytest_subprocess.FakeProcess, mocker
+):
+    def fake_execute(cmd: str):
+        return subprocess.check_output(  # noqa: S602
+            cmd,
+            shell=True,
+        )
+
+    def exec_fail(x):
+        raise subprocess.CalledProcessError(1, x)
+
+    fake_process.register(
+        ["cargo", "--version"], stdout="cargo 1.84.0-nightly (69e595908 2024-11-16)"
+    )
+    fake_process.register(
+        ["rustc", "--version"], stdout="rustc 1.84.0-nightly (5ec7d6eee 2024-11-17)"
+    )
+    fake_process.register(["rustup", "dump-testament"], callback=exec_fail)
+
+    mocker.patch(
+        "craft_parts.plugins.validator.PluginEnvironmentValidator._execute",
+        side_effect=fake_execute,
+    )
+    properties = RustPlugin.properties_class.unmarshal({"source": ".", "after": after})
+    validator = RustPlugin.validator_class(
+        part_name="my-part", properties=properties, env=""
+    )
+    # positive test
+    validator.validate_environment(part_dependencies=["rust-deps"])
+    # negative test
+    with pytest.raises(PluginEnvironmentValidationError):
+        validator.validate_environment(part_dependencies=after)
+
+
+def test_validate_environment_should_not_have_rustup(
+    fake_process: pytest_subprocess.FakeProcess, mocker
+):
+    def fake_execute(cmd: str):
+        return subprocess.check_output(  # noqa: S602
+            cmd,
+            shell=True,
+        )
+
+    def exec_fail(x):
+        raise subprocess.CalledProcessError(1, x)
```

Comment:
This should raise a `FileNotFoundError` instead:

```
Python 3.12.7 (main, Nov  8 2024, 17:55:36) [GCC 14.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import subprocess
>>> subproces.run(["hotunseaohuteonasuhoetnsuah"])
Traceback (most recent call last):
File "<stdin>", line 1, in <module>
NameError: name 'subproces' is not defined. Did you mean: 'subprocess'?
>>> subprocess.run(["hotunseaohuteonasuhoetnsuah"])
Traceback (most recent call last):
File "<stdin>", line 1, in <module>
File "/usr/lib/python3.12/subprocess.py", line 548, in run
with Popen(*popenargs, **kwargs) as process:
^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/lib/python3.12/subprocess.py", line 1026, in __init__
self._execute_child(args, executable, preexec_fn, close_fds,
File "/usr/lib/python3.12/subprocess.py", line 1955, in _execute_child
raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'hotunseaohuteonasuhoetnsuah'
```

Reply:
This is somewhat different, as `PluginEnvironmentValidator` uses `bash` to execute the commands: https://github.com/canonical/craft-parts/blob/a3b9cdab1853671eff69f8bf19cab89c27ee2aba/craft_parts/plugins/validator.py#L140-L150, so command not found still raises `CalledProcessError` but with `127` as the exit code.

Reply:
Ahh, thank you! I see you've changed it to `127` as well so we're all good on this piece.